thirdparty/runtime_environments.go

```diff
@@ -308,3 +311,89 @@ func (c *RuntimeEnvironmentsClient) getHistory(ctx context.Context, opts DistroH
 	}
 	return amiHistory, nil
 }
+
+// ImageEventEntry represents a change to the image.
+type ImageEventEntry struct {
+	Name   string
+	After  string
+	Before string
+	Type   string
+	Action string
+}
+
+// ImageEvent contains information about changes to an image when the ami changes.
+type ImageEvent struct {
+	Entries   []ImageEventEntry
+	Timestamp string
+	AMIBefore string
+	AMIAfter  string
+}
+
+// EventHistoryOptions represents the filtering arguments for getEvents. Distro is a required argument.
+type EventHistoryOptions struct {
+	Distro string
+	Page   int
+	Limit  int
+}
+
+// getEvents returns information about the changes between AMIs that occurred on the image.
+func (c *RuntimeEnvironmentsClient) getEvents(ctx context.Context, opts EventHistoryOptions) ([]ImageEvent, error) {
+	if opts.Distro == "" {
+		return nil, errors.Errorf("no distro provided")
+	}
+	limit := 10
+	if opts.Limit != 0 {
+		limit = opts.Limit
+	}
+	optsHistory := DistroHistoryFilterOptions{
+		Distro: opts.Distro,
+		Page:   opts.Page,
+		Limit:  limit + 1,
+	}
+	imageHistory, err := c.getHistory(ctx, optsHistory)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting image history")
+	}
+	result := []ImageEvent{}
+	for i, imageHistoryEntry := range imageHistory {
+		if i < len(imageHistory)-1 {
+			amiBefore := imageHistory[i+1].AMI
+			optsImageDiffs := ImageDiffOptions{
+				BeforeAMI: amiBefore,
+				AfterAMI:  imageHistoryEntry.AMI,
+			}
+			imageDiffs, err := c.getImageDiff(ctx, optsImageDiffs)
+			if err != nil {
+				return nil, errors.Wrap(err, "getting image differences")
+			}
+			entries := []ImageEventEntry{}
+			for _, diff := range imageDiffs {
+				action := ""
+				if diff.Added != "" && diff.Removed != "" {
+					action = UpdatedImageEntryAction
+				} else if diff.Added != "" {
+					action = AddedImageEntryAction
+				} else if diff.Removed != "" {
+					action = DeletedImageEntryAction
+				}
+				entry := ImageEventEntry{
+					Name:   diff.Name,
+					After:  diff.Added,
+					Before: diff.Removed,
+					Type:   diff.Type,
+					Action: action,
+				}
```

Comment:
possibly want to extract this info a helper function like `buildImageEventEntry`